lib/modules/manager/sbt/extract.ts

```diff
@@ -335,11 +385,121 @@ export function extractPackageFile(
     return null;
   }
 
-  const { deps, packageFileVersion } = parsedResult;
+  return parsedResult;
+}
 
-  if (!deps.length) {
-    return null;
+function prepareLoadPackageFiles(
+  packageFilesContent: { packageFile: string; content: string }[]
+): {
+  globalVars: ParseOptions['globalVars'];
+  registryUrls: string[];
+  scalaVersion: ParseOptions['scalaVersion'];
+} {
+  // Return variable
+  let globalVars: Variables = {};
+  let registryUrls: string[] = [REGISTRY_URLS.mavenCentral];
+  let scalaVersion: string | undefined = undefined;
+
+  for (const { packageFile, content } of packageFilesContent) {
+    const acc: PackageFile & ParseOptions = {
+      deps: [],
+      registryUrls,
+      localVars: globalVars,
+      globalVars: {},
+      packageFile,
+    };
+    const res = extractPackageFile(content, acc);
+
+    if (res) {
+      globalVars = { ...globalVars, ...res.localVars };
+      if (res.registryUrls) {
+        registryUrls = [...registryUrls, ...new Set(res.registryUrls)];
```

Comment:
Shouldn't pushing to the `registryUrls` array be enough? Deduplication also happens later on in `extractAllPackageFiles`

Reply:
i fixed this